RTSS discoverFunctionDependencies issue

#### System Information
- Ogre Version: Master
- Operating System / Platform: Android / IOS
- RenderSystem: GLSLES2

#### Detailled description
Computed RTSS shader miss some functions.
When I debug the program that load the shader dependencies, the "missing" functions seems to be imported correctly in the map.

but in GLSLESProgramWriter::discoverFunctionDependencies some functions are not found.
#### Ogre.log
03-27 16:09:57.407 29463 29481 I OGRE    : Creating resource group General
03-27 16:09:57.407 29463 29481 I OGRE    : Creating resource group OgreInternal
03-27 16:09:57.409 29463 29481 I OGRE    : Creating resource group OgreAutodetect
03-27 16:09:57.414 29463 29481 I OGRE    : SceneManagerFactory for type 'DefaultSceneManager' registered.
03-27 16:09:57.414 29463 29481 I OGRE    : Registering ResourceManager for type Material
03-27 16:09:57.415 29463 29481 I OGRE    : Registering ResourceManager for type Mesh
03-27 16:09:57.415 29463 29481 I OGRE    : Registering ResourceManager for type Skeleton
03-27 16:09:57.415 29463 29481 I OGRE    : MovableObjectFactory for type 'ParticleSystem' registered.
03-27 16:09:57.415 29463 29481 I OGRE    : ArchiveFactory for archive type FileSystem registered.
03-27 16:09:57.417 29463 29481 I OGRE    : ArchiveFactory for archive type Zip registered.
03-27 16:09:57.417 29463 29481 I OGRE    : ArchiveFactory for archive type EmbeddedZip registered.
03-27 16:09:57.417 29463 29481 I OGRE    : DDS codec registering
03-27 16:09:57.417 29463 29481 I OGRE    : PVRTC codec registering
03-27 16:09:57.417 29463 29481 I OGRE    : ETC codec registering
03-27 16:09:57.417 29463 29481 I OGRE    : Registering ResourceManager for type HighLevelGpuProgram
03-27 16:09:57.418 29463 29481 I OGRE    : Registering ResourceManager for type Compositor
03-27 16:09:57.418 29463 29481 I OGRE    : MovableObjectFactory for type 'Entity' registered.
03-27 16:09:57.419 29463 29481 I OGRE    : MovableObjectFactory for type 'Light' registered.
03-27 16:09:57.419 29463 29481 I OGRE    : MovableObjectFactory for type 'BillboardSet' registered.
03-27 16:09:57.419 29463 29481 I OGRE    : MovableObjectFactory for type 'ManualObject' registered.
03-27 16:09:57.419 29463 29481 I OGRE    : MovableObjectFactory for type 'BillboardChain' registered.
03-27 16:09:57.419 29463 29481 I OGRE    : MovableObjectFactory for type 'RibbonTrail' registered.
03-27 16:09:57.419 29463 29481 I OGRE    : *-*-* OGRE Initialising
03-27 16:09:57.419 29463 29481 I OGRE    : *-*-* Version 1.12.0 (Rhagorthua)
03-27 16:09:57.419 29463 29481 I OGRE    : Creating resource group SO3/Internal/Widgets/BitmapWidget
03-27 16:09:57.419 29463 29481 I OGRE    : Installing plugin: OpenGL ES 2.0 RenderSystem
03-27 16:09:57.420 29463 29481 I OGRE    : OpenGL ES 2.x Rendering Subsystem created.
03-27 16:09:57.470 29463 29481 I OGRE    : Plugin successfully installed
03-27 16:09:57.470 29463 29481 I OGRE    : Installing plugin: ParticleFX
03-27 16:09:57.470 29463 29481 I OGRE    : Particle Emitter Type 'Point' registered
03-27 16:09:57.470 29463 29481 I OGRE    : Particle Emitter Type 'Box' registered
03-27 16:09:57.470 29463 29481 I OGRE    : Particle Emitter Type 'Ellipsoid' registered
03-27 16:09:57.470 29463 29481 I OGRE    : Particle Emitter Type 'Cylinder' registered
03-27 16:09:57.470 29463 29481 I OGRE    : Particle Emitter Type 'Ring' registered
03-27 16:09:57.471 29463 29481 I OGRE    : Particle Emitter Type 'HollowEllipsoid' registered
03-27 16:09:57.471 29463 29481 I OGRE    : Particle Affector Type 'LinearForce' registered
03-27 16:09:57.471 29463 29481 I OGRE    : Particle Affector Type 'ColourFader' registered
03-27 16:09:57.471 29463 29481 I OGRE    : Particle Affector Type 'ColourFader2' registered
03-27 16:09:57.471 29463 29481 I OGRE    : Particle Affector Type 'ColourImage' registered
03-27 16:09:57.471 29463 29481 I OGRE    : Particle Affector Type 'ColourInterpolator' registered
03-27 16:09:57.471 29463 29481 I OGRE    : Particle Affector Type 'Scaler' registered
03-27 16:09:57.471 29463 29481 I OGRE    : Particle Affector Type 'Rotator' registered
03-27 16:09:57.472 29463 29481 I OGRE    : Particle Affector Type 'DirectionRandomiser' registered
03-27 16:09:57.472 29463 29481 I OGRE    : Particle Affector Type 'DeflectorPlane' registered
03-27 16:09:57.472 29463 29481 I OGRE    : Plugin successfully installed
03-27 16:09:57.472 29463 29481 I OGRE    : Installing plugin: STB Image Codec
03-27 16:09:57.472 29463 29481 I OGRE    : stb_image - v2.19 - public domain JPEG/PNG reader
03-27 16:09:57.472 29463 29481 I OGRE    : Supported formats: jpeg,jpg,png,bmp,psd,tga,gif,pic,ppm,pgm,hdr
03-27 16:09:57.472 29463 29481 I OGRE    : Plugin successfully installed
03-27 16:09:57.473 29463 29481 I OGRE    : Installing plugin: Octree Scene Manager
03-27 16:09:57.473 29463 29481 I OGRE    : Plugin successfully installed
03-27 16:09:57.473 29463 29481 I OGRE    : ArchiveFactory for archive type ScolFileSystem registered.
03-27 16:09:57.473 29463 29481 I OGRE    : ArchiveFactory for archive type ScolURL registered.
03-27 16:09:57.473 29463 29481 I OGRE    : ArchiveFactory for archive type APKFileSystem registered.
03-27 16:09:57.473 29463 29481 I OGRE    : ArchiveFactory for archive type APKZip registered.
03-27 16:09:57.475 29463 29481 I OGRE    : CPU Identifier & Features
03-27 16:09:57.475 29463 29481 I OGRE    : -------------------------
03-27 16:09:57.477 29463 29481 I OGRE    :  *   CPU ID: ARMv7
03-27 16:09:57.477 29463 29481 I OGRE    :  *          VFP: yes
03-27 16:09:57.478 29463 29481 I OGRE    :  *         NEON: yes
03-27 16:09:57.478 29463 29481 I OGRE    : -------------------------
03-27 16:09:57.479 29463 29481 I OGRE    : ******************************
03-27 16:09:57.479 29463 29481 I OGRE    : *** Starting EGL Subsystem ***
03-27 16:09:57.479 29463 29481 I OGRE    : ******************************
03-27 16:09:57.537 29463 29481 I OGRE    : EGL_VERSION = 1.4 Android META-EGL
03-27 16:09:57.552 29463 29481 I OGRE    : EGL_EXTENSIONS = EGL_KHR_get_all_proc_addresses EGL_ANDROID_presentation_time EGL_KHR_swap_buffers_with_damage EGL_ANDROID_create_native_client_buffer EGL_ANDROID_front_buffer_auto_refresh EGL_KHR_image EGL_KHR_image_base EGL_KHR_gl_colorspace EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_gl_renderbuffer_image EGL_KHR_reusable_sync EGL_KHR_fence_sync EGL_KHR_create_context EGL_KHR_config_attribs EGL_KHR_surfaceless_context EGL_KHR_stream EGL_KHR_stream_fifo EGL_KHR_stream_producer_eglsurface EGL_KHR_stream_consumer_gltexture EGL_KHR_stream_cross_process_fd EGL_EXT_create_context_robustness EGL_NV_system_time EGL_NV_sync EGL_NV_stream_sync EGL_IMG_context_priority EGL_ANDROID_image_native_buffer EGL_KHR_wait_sync EGL_ANDROID_recordable EGL_KHR_partial_update EGL_EXT_buffer_age EGL_KHR_create_context_no_error EGL_KHR_mutable_render_buffer EGL_EXT_protected_content 
03-27 16:09:57.552 29463 29481 I OGRE    : Registering ResourceManager for type Texture
03-27 16:09:57.552 29463 29481 I OGRE    : OverlayElementFactory for type Panel registered.
03-27 16:09:57.553 29463 29481 I OGRE    : OverlayElementFactory for type BorderPanel registered.
03-27 16:09:57.553 29463 29481 I OGRE    : OverlayElementFactory for type TextArea registered.
03-27 16:09:57.553 29463 29481 I OGRE    : Registering ResourceManager for type Font
03-27 16:10:04.344 29463 29481 I OGRE    : GLES2RenderSystem::_createRenderWindow "SCOL_buffer_1", 0x0 windowed  miscParams: androidConfig=-1470975504 contentScalingFactor=1.5 externalWindowHandle=-1534535672 minColourBufferSize=32 
03-27 16:10:04.367 29463 29481 I OGRE    : GL_VERSION = 3.2.0.0
03-27 16:10:04.367 29463 29481 I OGRE    : GL_VENDOR = NVIDIA Corporation
03-27 16:10:04.367 29463 29481 I OGRE    : GL_RENDERER = NVIDIA Tegra
03-27 16:10:04.368 29463 29481 I OGRE    : GL_EXTENSIONS = GL_EXT_debug_marker GL_EXT_base_instance GL_EXT_blend_func_extended GL_EXT_blend_minmax GL_EXT_buffer_storage GL_EXT_clip_cull_distance GL_EXT_color_buffer_float GL_EXT_color_buffer_half_float GL_EXT_copy_image GL_EXT_debug_label GL_EXT_discard_framebuffer GL_EXT_disjoint_timer_query GL_EXT_draw_buffers_indexed GL_EXT_draw_elements_base_vertex GL_EXT_float_blend GL_EXT_frag_depth GL_EXT_geometry_point_size GL_EXT_geometry_shader GL_EXT_gpu_shader5 GL_EXT_map_buffer_range GL_EXT_multi_draw_indirect GL_EXT_multisample_compatibility GL_EXT_occlusion_query_boolean GL_EXT_polygon_offset_clamp GL_EXT_primitive_bounding_box GL_EXT_protected_textures GL_EXT_render_snorm GL_EXT_robustness GL_EXT_separate_shader_objects GL_EXT_shader_implicit_conversions GL_EXT_shader_integer_mix GL_EXT_shader_io_blocks GL_EXT_shader_non_constant_global_initializers GL_EXT_shader_texture_lod GL_EXT_shadow_samplers GL_EXT_sparse_texture GL_EXT_sRGB GL_EXT_sRGB_write_control GL_EXT_tessellation_point_size GL_EXT_tessella
03-27 16:10:04.369 29463 29481 I OGRE    : **************************************
03-27 16:10:04.369 29463 29481 I OGRE    : *** OpenGL ES 2.x Renderer Started ***
03-27 16:10:04.369 29463 29481 I OGRE    : **************************************
03-27 16:10:04.370 29463 29481 I OGRE    : Shading language version: OpenGL ES GLSL ES 3.20
03-27 16:10:04.370 29463 29481 I OGRE    : Registering ResourceManager for type GpuProgram
03-27 16:10:04.370 29463 29481 I OGRE    : GL ES 2: Using FBOs for rendering to textures
03-27 16:10:04.377 29463 29481 I OGRE    : FBO PF_L8 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.382 29463 29481 I OGRE    : FBO PF_BYTE_LA depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.388 29463 29481 I OGRE    : FBO PF_R5G6B5 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.393 29463 29481 I OGRE    : FBO PF_A4R4G4B4 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.398 29463 29481 I OGRE    : FBO PF_A1R5G5B5 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.404 29463 29481 I OGRE    : FBO PF_B8G8R8 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.409 29463 29481 I OGRE    : FBO PF_A8B8G8R8 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.414 29463 29481 I OGRE    : FBO PF_A2B10G10R10 depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.420 29463 29481 I OGRE    : FBO PF_FLOAT16_RGB depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.426 29463 29481 I OGRE    : FBO PF_FLOAT16_RGBA depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.432 29463 29481 I OGRE    : FBO PF_FLOAT32_RGBA depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.438 29463 29481 I OGRE    : FBO PF_X8B8G8R8 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.446 29463 29481 I OGRE    : FBO PF_DEPTH16 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.452 29463 29481 I OGRE    : FBO PF_FLOAT16_R depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.457 29463 29481 I OGRE    : FBO PF_FLOAT32_R depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.464 29463 29481 I OGRE    : FBO PF_FLOAT16_GR depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.471 29463 29481 I OGRE    : FBO PF_FLOAT32_GR depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.476 29463 29481 I OGRE    : FBO PF_R11G11B10_FLOAT depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.482 29463 29481 I OGRE    : FBO PF_R8_UINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.487 29463 29481 I OGRE    : FBO PF_R8G8_UINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.493 29463 29481 I OGRE    : FBO PF_R8G8B8A8_UINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.498 29463 29481 I OGRE    : FBO PF_R16_UINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.504 29463 29481 I OGRE    : FBO PF_R16G16_UINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.510 29463 29481 I OGRE    : FBO PF_R16G16B16A16_UINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.516 29463 29481 I OGRE    : FBO PF_R32_UINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.522 29463 29481 I OGRE    : FBO PF_R32G32_UINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.528 29463 29481 I OGRE    : FBO PF_R32G32B32A32_UINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.534 29463 29481 I OGRE    : FBO PF_R8_SINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.540 29463 29481 I OGRE    : FBO PF_R8G8_SINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.547 29463 29481 I OGRE    : FBO PF_R8G8B8A8_SINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.557 29463 29481 I OGRE    : FBO PF_R16_SINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.663 29463 29481 I OGRE    : FBO PF_R16G16_SINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.668 29463 29481 I OGRE    : FBO PF_R16G16B16A16_SINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.673 29463 29481 I OGRE    : FBO PF_R32_SINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.677 29463 29481 I OGRE    : FBO PF_R32G32_SINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.682 29463 29481 I OGRE    : FBO PF_R32G32B32A32_SINT depth/stencil support: D0S0 D0S8 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.687 29463 29481 I OGRE    : FBO PF_R8 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.691 29463 29481 I OGRE    : FBO PF_RG8 depth/stencil support: D0S0 D0S8 D16S0 D24S0 D32S0 Packed-D24S8 Packed-D32S8 
03-27 16:10:04.692 29463 29481 I OGRE    : [GLES2] : Valid FBO targets PF_L8 PF_BYTE_LA PF_R5G6B5 PF_A4R4G4B4 PF_A1R5G5B5 PF_B8G8R8 PF_A8B8G8R8 PF_A2B10G10R10 PF_FLOAT16_RGB PF_FLOAT16_RGBA PF_FLOAT32_RGBA PF_X8B8G8R8 PF_DEPTH16 PF_FLOAT16_R PF_FLOAT32_R PF_FLOAT16_GR PF_FLOAT32_GR PF_R11G11B10_FLOAT PF_R8_UINT PF_R8G8_UINT PF_R8G8B8A8_UINT PF_R16_UINT PF_R16G16_UINT PF_R16G16B16A16_UINT PF_R32_UINT PF_R32G32_UINT PF_R32G32B32A32_UINT PF_R8_SINT PF_R8G8_SINT PF_R8G8B8A8_SINT PF_R16_SINT PF_R16G16_SINT PF_R16G16B16A16_SINT PF_R32_SINT PF_R32G32_SINT PF_R32G32B32A32_SINT PF_R8 PF_RG8 
03-27 16:10:04.692 29463 29481 I OGRE    : RenderSystem capabilities
03-27 16:10:04.692 29463 29481 I OGRE    : -------------------------
03-27 16:10:04.693 29463 29481 I OGRE    : RenderSystem Name: OpenGL ES 2.x Rendering Subsystem
03-27 16:10:04.693 29463 29481 I OGRE    : GPU Vendor: nvidia
03-27 16:10:04.693 29463 29481 I OGRE    : Device Name: NVIDIA Tegra
03-27 16:10:04.693 29463 29481 I OGRE    : Driver Version: 3.2.0.0
03-27 16:10:04.693 29463 29481 I OGRE    :  * Fixed function pipeline: no
03-27 16:10:04.693 29463 29481 I OGRE    :  * Anisotropic texture filtering: yes
03-27 16:10:04.693 29463 29481 I OGRE    :  * Hardware stencil buffer: no
03-27 16:10:04.693 29463 29481 I OGRE    :  * 32-bit index buffers: yes
03-27 16:10:04.693 29463 29481 I OGRE    :  * Vertex programs: yes
03-27 16:10:04.693 29463 29481 I OGRE    :  * Number of floating-point constants for vertex programs: 1024
03-27 16:10:04.694 29463 29481 I OGRE    :  * Number of integer constants for vertex programs: 1024
03-27 16:10:04.694 29463 29481 I OGRE    :  * Number of boolean constants for vertex programs: 1024
03-27 16:10:04.694 29463 29481 I OGRE    :  * Fragment programs: yes
03-27 16:10:04.694 29463 29481 I OGRE    :  * Number of floating-point constants for fragment programs: 1024
03-27 16:10:04.694 29463 29481 I OGRE    :  * Number of integer constants for fragment programs: 1024
03-27 16:10:04.694 29463 29481 I OGRE    :  * Number of boolean constants for fragment programs: 1024
03-27 16:10:04.694 29463 29481 I OGRE    :  * Geometry programs: no
03-27 16:10:04.694 29463 29481 I OGRE    :  * Number of floating-point constants for geometry programs: 0
03-27 16:10:04.694 29463 29481 I OGRE    :  * Number of integer constants for geometry programs: 0
03-27 16:10:04.694 29463 29481 I OGRE    :  * Number of boolean constants for geometry programs: 0
03-27 16:10:04.695 29463 29481 I OGRE    :  * Tessellation Hull programs: no
03-27 16:10:04.695 29463 29481 I OGRE    :  * Number of floating-point constants for tessellation hull programs: 0
03-27 16:10:04.695 29463 29481 I OGRE    :  * Number of integer constants for tessellation hull programs: 0
03-27 16:10:04.695 29463 29481 I OGRE    :  * Number of boolean constants for tessellation hull programs: 0
03-27 16:10:04.695 29463 29481 I OGRE    :  * Tessellation Domain programs: no
03-27 16:10:04.695 29463 29481 I OGRE    :  * Number of floating-point constants for tessellation domain programs: 0
03-27 16:10:04.695 29463 29481 I OGRE    :  * Number of integer constants for tessellation domain programs: 0
03-27 16:10:04.695 29463 29481 I OGRE    :  * Number of boolean constants for tessellation domain programs: 0
03-27 16:10:04.695 29463 29481 I OGRE    :  * Compute programs: no
03-27 16:10:04.695 29463 29481 I OGRE    :  * Number of floating-point constants for compute programs: 0
03-27 16:10:04.695 29463 29481 I OGRE    :  * Number of integer constants for compute programs: 0
03-27 16:10:04.696 29463 29481 I OGRE    :  * Number of boolean constants for compute programs: 0
03-27 16:10:04.696 29463 29481 I OGRE    :  * Supported Shader Profiles: glsl300es glsl310es glsl320es glsles
03-27 16:10:04.696 29463 29481 I OGRE    :  * Texture Compression: yes
03-27 16:10:04.696 29463 29481 I OGRE    :    - DXT: yes
03-27 16:10:04.696 29463 29481 I OGRE    :    - VTC: no
03-27 16:10:04.696 29463 29481 I OGRE    :    - PVRTC: no
03-27 16:10:04.696 29463 29481 I OGRE    :    - ATC: no
03-27 16:10:04.696 29463 29481 I OGRE    :    - ETC1: yes
03-27 16:10:04.696 29463 29481 I OGRE    :    - ETC2: yes
03-27 16:10:04.696 29463 29481 I OGRE    :    - BC4/BC5: no
03-27 16:10:04.697 29463 29481 I OGRE    :    - BC6H/BC7: no
03-27 16:10:04.697 29463 29481 I OGRE    :    - ASTC: yes
03-27 16:10:04.697 29463 29481 I OGRE    :    - Mipmaps for compressed formats: no
03-27 16:10:04.697 29463 29481 I OGRE    :  * Hardware Occlusion Query: yes
03-27 16:10:04.697 29463 29481 I OGRE    :  * User clip planes: no
03-27 16:10:04.697 29463 29481 I OGRE    :  * VET_UBYTE4 vertex element type: yes
03-27 16:10:04.697 29463 29481 I OGRE    :  * Infinite far plane projection: yes
03-27 16:10:04.697 29463 29481 I OGRE    :  * Hardware render-to-texture: yes
03-27 16:10:04.697 29463 29481 I OGRE    :  * Floating point textures: yes
03-27 16:10:04.697 29463 29481 I OGRE    :  * Non-power-of-two textures: yes
03-27 16:10:04.697 29463 29481 I OGRE    :  * 1d textures: no
03-27 16:10:04.697 29463 29481 I OGRE    :  * Volume textures: yes
03-27 16:10:04.698 29463 29481 I OGRE    :  * Multiple Render Targets: 8
03-27 16:10:04.698 29463 29481 I OGRE    :    - With different bit depths: yes
03-27 16:10:04.698 29463 29481 I OGRE    :  * Point Sprites: yes
03-27 16:10:04.698 29463 29481 I OGRE    :  * Wide Lines: yes
03-27 16:10:04.698 29463 29481 I OGRE    :  * Hardware Gamma: yes
03-27 16:10:04.698 29463 29481 I OGRE    :  * Extended point parameters: yes
03-27 16:10:04.698 29463 29481 I OGRE    :  * Max Point Size: 0
03-27 16:10:04.698 29463 29481 I OGRE    :  * Vertex texture fetch: no
03-27 16:10:04.698 29463 29481 I OGRE    :  * Number of texture units: 16
03-27 16:10:04.698 29463 29481 I OGRE    :  * Number of vertex attributes: 16
03-27 16:10:04.698 29463 29481 I OGRE    :  * Stencil buffer depth: 0
03-27 16:10:04.698 29463 29481 I OGRE    :  * Number of vertex blend matrices: 0
03-27 16:10:04.699 29463 29481 I OGRE    :  * Render to Vertex Buffer : yes
03-27 16:10:04.699 29463 29481 I OGRE    :  * Hardware Atomic Counters: no
03-27 16:10:04.699 29463 29481 I OGRE    :  * PBuffer support: no
03-27 16:10:04.699 29463 29481 I OGRE    :  * Vertex Array Objects: yes
03-27 16:10:04.699 29463 29481 I OGRE    :  * Separate shader objects: yes
03-27 16:10:04.699 29463 29481 I OGRE    :  * GLSL SSO redeclare interface block: yes
03-27 16:10:04.699 29463 29481 I OGRE    :  * Debugging/ profiling events: yes
03-27 16:10:04.699 29463 29481 I OGRE    :  * Map buffer storage: yes
03-27 16:10:04.699 29463 29481 I OGRE    :  * Quad Buffer: no
03-27 16:10:04.699 29463 29481 I OGRE    : DefaultWorkQueue('Root') initialising on thread -1467582176.
03-27 16:10:04.700 29463 29481 I OGRE    : Particle Renderer Type 'billboard' registered
03-27 16:10:04.700 29463 29489 I OGRE    : DefaultWorkQueue('Root')::WorkerFunc - thread -1911551712 starting.
03-27 16:10:04.701 29463 29481 I OGRE    : SceneManagerFactory for type 'OctreeSceneManager' registered.
03-27 16:10:04.701 29463 29481 I OGRE    : Creating resource group SO3/Internal
03-27 16:10:04.701 29463 29481 I OGRE    : Creating resource group SO3/Internal/DynResource
03-27 16:10:04.701 29463 29481 I OGRE    : Creating resource group SO3/Internal/GlobalPool
03-27 16:10:04.701 29463 29481 I OGRE    : Creating resource group Hydrax
03-27 16:10:04.701 29463 29481 I OGRE    : Creating resource group SkyX
03-27 16:10:04.701 29463 29481 I OGRE    : Creating resource group SO3/Internal/EntitySelector
03-27 16:10:04.704 29463 29481 I OGRE    : Added resource location 'ogre_redist/ogrecore.zip' of type 'APKZip' to resource group 'OgreInternal' with recursive option
03-27 16:10:04.704 29463 29481 I OGRE    : Added resource location 'ogre_redist/rtshaderlib.zip' of type 'APKZip' to resource group 'OgreInternal' with recursive option
03-27 16:10:04.705 29463 29488 I OGRE    : DefaultWorkQueue('Root')::WorkerFunc - thread -1910507232 starting.
03-27 16:10:04.708 29463 29481 I OGRE    : Added resource location 'ogre_redist/so3core.zip' of type 'APKZip' to resource group 'SO3/Internal' with recursive option
03-27 16:10:04.708 29463 29481 I OGRE    : Added resource location 'ogre_redist/hydrax.zip' of type 'APKZip' to resource group 'Hydrax' with recursive option
03-27 16:10:04.709 29463 29481 I OGRE    : Added resource location 'ogre_redist/skyx.zip' of type 'APKZip' to resource group 'SkyX' with recursive option
03-27 16:10:04.710 29463 29481 I OGRE    : Added resource location 'ogre_redist/so3selectionbuffer.zip' of type 'APKZip' to resource group 'SO3/Internal/EntitySelector' with recursive option
03-27 16:10:04.710 29463 29481 I OGRE    : Added resource location 'ogre_redist/fonts.zip' of type 'APKZip' to resource group 'SO3/Internal' with recursive option
03-27 16:10:04.711 29463 29481 I OGRE    : Parsing scripts for resource group General
03-27 16:10:04.711 29463 29481 I OGRE    : Finished parsing scripts for resource group General
03-27 16:10:04.711 29463 29481 I OGRE    : Creating resources for group General
03-27 16:10:04.711 29463 29481 I OGRE    : All done
03-27 16:10:04.711 29463 29481 I OGRE    : Parsing scripts for resource group Hydrax
03-27 16:10:04.712 29463 29481 I OGRE    : Finished parsing scripts for resource group Hydrax
03-27 16:10:04.712 29463 29481 I OGRE    : Creating resources for group Hydrax
03-27 16:10:04.713 29463 29481 I OGRE    : All done
03-27 16:10:04.713 29463 29481 I OGRE    : Parsing scripts for resource group OgreAutodetect
03-27 16:10:04.713 29463 29481 I OGRE    : Finished parsing scripts for resource group OgreAutodetect
03-27 16:10:04.713 29463 29481 I OGRE    : Creating resources for group OgreAutodetect
03-27 16:10:04.713 29463 29481 I OGRE    : All done
03-27 16:10:04.713 29463 29481 I OGRE    : Parsing scripts for resource group OgreInternal
03-27 16:10:04.716 29463 29481 I OGRE    : Parsing script ShadowVolumeExtude.program
03-27 16:10:04.720 29463 29481 I OGRE    : Parsing script OgreProfiler.material
03-27 16:10:04.723 29463 29481 I OGRE    : Parsing script SdkTrays.material
03-27 16:10:04.731 29463 29481 I OGRE    : Parsing script HardwareSkinningShadow.material
03-27 16:10:04.738 29463 29481 I OGRE    : Parsing script RTShaderSystem.material
03-27 16:10:04.743 29463 29481 I OGRE    : Parsing script TriplanarTexturing.material
03-27 16:10:04.744 29463 29481 I OGRE    : Parsing script DualQuaternionSkinning_Shadow.material
....
03-27 16:10:04.805 29463 29481 I OGRE    : Finished parsing scripts for resource group OgreInternal
03-27 16:10:04.805 29463 29481 I OGRE    : Creating resources for group OgreInternal
03-27 16:10:04.806 29463 29481 I OGRE    : All done
03-27 16:10:04.806 29463 29481 I OGRE    : Parsing scripts for resource group SO3/Internal
03-27 16:10:04.811 29463 29481 I OGRE    : Parsing script StdQuad_vp.program
03-27 16:10:04.818 29463 29481 I OGRE    : Parsing script ShadowReceiver.program
03-27 16:10:04.834 29463 29481 I OGRE    : Parsing script ShadowCaster.program
03-27 16:10:04.836 29463 29481 I OGRE    : Parsing script overlay.program
03-27 16:10:04.838 29463 29481 I OGRE    : Parsing script linecolor.program
03-27 16:10:04.840 29463 29481 I OGRE    : Parsing script SecurityCam.program
03-27 16:10:04.845 29463 29481 I OGRE    : Parsing script linecustomcolor.program
03-27 16:10:04.847 29463 29481 I OGRE    : Parsing script lispsm.program
03-27 16:10:04.853 29463 29481 I OGRE    : Parsing script pssm.program
03-27 16:10:04.864 29463 29481 I OGRE    : Parsing script oculus.material
03-27 16:10:04.866 29463 29481 I OGRE    : Parsing script oculus2.material
03-27 16:10:04.868 29463 29481 I OGRE    : Parsing script Stereoscopy.material
03-27 16:10:04.956 29463 29481 I OGRE    : Parsing script LigthShafts.material
03-27 16:10:04.962 29463 29481 I OGRE    : Parsing script ShadowReceiver.material
03-27 16:10:04.970 29463 29481 I OGRE    : Parsing script default.material
03-27 16:10:04.975 29463 29481 I OGRE    : Parsing script ASCII.material
03-27 16:10:04.979 29463 29481 I OGRE    : Parsing script BlackAndWhite.material
03-27 16:10:04.982 29463 29481 I OGRE    : Parsing script Bloom.material
03-27 16:10:04.988 29463 29481 I OGRE    : Parsing script Bloom2.material
03-27 16:10:04.997 29463 29481 I OGRE    : Parsing script DOF.material
03-27 16:10:05.000 29463 29481 I OGRE    : Parsing script Embossed.material
03-27 16:10:05.002 29463 29481 I OGRE    : Parsing script glass.material
03-27 16:10:05.005 29463 29481 I OGRE    : Parsing script HeatVision.material
03-27 16:10:05.015 29463 29481 I OGRE    : Parsing script Invert.material
03-27 16:10:05.018 29463 29481 I OGRE    : Parsing script Laplace.material
03-27 16:10:05.020 29463 29481 I OGRE    : Parsing script NightVision.material
03-27 16:10:05.024 29463 29481 I OGRE    : Parsing script OldMovie.material
03-27 16:10:05.028 29463 29481 I OGRE    : Parsing script OldTV.material
03-27 16:10:05.032 29463 29481 I OGRE    : Parsing script Posterize.material
03-27 16:10:05.035 29463 29481 I OGRE    : Parsing script RadialBlur.material
03-27 16:10:05.037 29463 29481 I OGRE    : Parsing script securitycam.material
03-27 16:10:05.040 29463 29481 I OGRE    : Parsing script SharpenEdges.material
03-27 16:10:05.043 29463 29481 I OGRE    : Parsing script Swizzle.material
03-27 16:10:05.045 29463 29481 I OGRE    : Parsing script Tiling.material
03-27 16:10:05.048 29463 29481 I OGRE    : Parsing script lispsm.material
03-27 16:10:05.051 29463 29481 I OGRE    : Parsing script pssm.material
03-27 16:10:05.057 29463 29481 I OGRE    : Parsing script oculus.compositor
03-27 16:10:05.058 29463 29481 I OGRE    : Parsing script Examples.compositor
03-27 16:10:05.065 29463 29481 I OGRE    : Parsing script securitycam.compositor
03-27 16:10:05.067 29463 29481 I OGRE    : Parsing script Ogre.fontdef
03-27 16:10:05.068 29463 29481 I OGRE    : Finished parsing scripts for resource group SO3/Internal
03-27 16:10:05.068 29463 29481 I OGRE    : Creating resources for group SO3/Internal
03-27 16:10:05.068 29463 29481 I OGRE    : All done
03-27 16:10:05.068 29463 29481 I OGRE    : Parsing scripts for resource group SO3/Internal/DynResource
03-27 16:10:05.068 29463 29481 I OGRE    : Finished parsing scripts for resource group SO3/Internal/DynResource
03-27 16:10:05.068 29463 29481 I OGRE    : Creating resources for group SO3/Internal/DynResource
03-27 16:10:05.068 29463 29481 I OGRE    : All done
03-27 16:10:05.068 29463 29481 I OGRE    : Parsing scripts for resource group SO3/Internal/EntitySelector
03-27 16:10:05.069 29463 29481 I OGRE    : Parsing script entitySelectorColor.program
03-27 16:10:05.080 29463 29481 I OGRE    : Parsing script entitySelectorColor.material
03-27 16:10:05.085 29463 29481 I OGRE    : Finished parsing scripts for resource group SO3/Internal/EntitySelector
03-27 16:10:05.085 29463 29481 I OGRE    : Creating resources for group SO3/Internal/EntitySelector
03-27 16:10:05.086 29463 29481 I OGRE    : All done
03-27 16:10:05.086 29463 29481 I OGRE    : Parsing scripts for resource group SO3/Internal/GlobalPool
03-27 16:10:05.086 29463 29481 I OGRE    : Finished parsing scripts for resource group SO3/Internal/GlobalPool
03-27 16:10:05.086 29463 29481 I OGRE    : Creating resources for group SO3/Internal/GlobalPool
03-27 16:10:05.086 29463 29481 I OGRE    : All done
03-27 16:10:05.086 29463 29481 I OGRE    : Parsing scripts for resource group SO3/Internal/Widgets/BitmapWidget
03-27 16:10:05.086 29463 29481 I OGRE    : Finished parsing scripts for resource group SO3/Internal/Widgets/BitmapWidget
03-27 16:10:05.086 29463 29481 I OGRE    : Creating resources for group SO3/Internal/Widgets/BitmapWidget
03-27 16:10:05.086 29463 29481 I OGRE    : All done
03-27 16:10:05.087 29463 29481 I OGRE    : Parsing scripts for resource group SkyX
03-27 16:10:05.088 29463 29481 I OGRE    : Parsing script SkyX.material
03-27 16:10:05.126 29463 29481 I OGRE    : Finished parsing scripts for resource group SkyX
03-27 16:10:05.126 29463 29481 I OGRE    : Creating resources for group SkyX
03-27 16:10:05.127 29463 29481 I OGRE    : All done
03-27 16:10:05.129 29463 29481 I OGRE    : Creating resource group SO3/Internal/EntitySelector/SCOL_buffer_1
03-27 16:10:05.129 29463 29481 I OGRE    : GLES2Texture::create - Name: SO3/Internal/EntitySelector/SCOL_buffer_1/Texture ID: 16 Width: 320 Height: 184 Internal Format: 0x8051
03-27 16:10:05.130 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture SO3/Internal/EntitySelector/SCOL_buffer_1/Texture id 16 face 0 level 0: width=320 height=184 depth=1 format=PF_B8G8R8
03-27 16:10:05.132 29463 29481 I OGRE    : GLES2Texture::create - Name: SO3/Internal/EntitySelector/SCOL_buffer_1/Picker ID: 17 Width: 1 Height: 1 Internal Format: 0x8051
03-27 16:10:05.133 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture SO3/Internal/EntitySelector/SCOL_buffer_1/Picker id 17 face 0 level 0: width=1 height=1 depth=1 format=PF_B8G8R8
03-27 16:10:05.138 29463 29481 I OGRE    : Creating resource group 0.mainGeneral
03-27 16:10:05.139 29463 29481 I OGRE    : Added resource location '/storage/emulated/0/Android/data/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainGeneral'
03-27 16:10:05.139 29463 29481 I OGRE    : Added resource location '/data/user/0/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainGeneral'
03-27 16:10:05.140 29463 29481 I OGRE    : [WARNING] ::find Could not find the files in APK with pattern : *
03-27 16:10:05.140 29463 29481 I OGRE    : Added resource location 'APK/partition/' of type 'ScolFileSystem' to resource group '0.mainGeneral'
03-27 16:10:05.140 29463 29481 I OGRE    : Added resource location 'OBB/partition/' of type 'ScolFileSystem' to resource group '0.mainGeneral'
03-27 16:10:05.142 29463 29481 I OGRE    : Creating resource group 0.mainV3DdefaultResources
03-27 16:10:05.143 29463 29481 I OGRE    : Added resource location '/storage/emulated/0/Android/data/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainV3DdefaultResources'
03-27 16:10:05.144 29463 29481 I OGRE    : Added resource location '/data/user/0/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainV3DdefaultResources'
03-27 16:10:05.144 29463 29481 I OGRE    : [WARNING] ::find Could not find the files in APK with pattern : *
03-27 16:10:05.144 29463 29481 I OGRE    : Added resource location 'APK/partition/' of type 'ScolFileSystem' to resource group '0.mainV3DdefaultResources'
03-27 16:10:05.144 29463 29481 I OGRE    : Added resource location 'OBB/partition/' of type 'ScolFileSystem' to resource group '0.mainV3DdefaultResources'
03-27 16:10:05.145 29463 29481 I OGRE    : GLES2Texture::create - Name: SO3/Internal/EntitySelector/SCOL_buffer_1/Texture ID: 16 Width: 320 Height: 184 Internal Format: 0x8051
03-27 16:10:05.145 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture SO3/Internal/EntitySelector/SCOL_buffer_1/Texture id 16 face 0 level 0: width=320 height=184 depth=1 format=PF_B8G8R8
03-27 16:10:05.193 29463 29481 I OGRE    : GLES2Texture::create - Name: VUI3D_6_3dconttex ID: 18 Width: 2 Height: 2 Internal Format: 0x8058
03-27 16:10:05.193 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_6_3dconttex id 18 face 0 level 0: width=2 height=2 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.194 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_6_3dconttex id 18 face 0 level 1: width=1 height=1 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.198 29463 29481 I OGRE    : GLES2Texture::create - Name: VUI3D_7_3dconttex ID: 19 Width: 2 Height: 2 Internal Format: 0x8058
03-27 16:10:05.198 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_7_3dconttex id 19 face 0 level 0: width=2 height=2 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.198 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_7_3dconttex id 19 face 0 level 1: width=1 height=1 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.201 29463 29481 I OGRE    : GLES2Texture::create - Name: VUI3D_8_3dconttex ID: 20 Width: 2 Height: 2 Internal Format: 0x8058
03-27 16:10:05.201 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_8_3dconttex id 20 face 0 level 0: width=2 height=2 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.201 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_8_3dconttex id 20 face 0 level 1: width=1 height=1 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.203 29463 29481 I OGRE    : GLES2Texture::create - Name: VUI3D_9_3dconttex ID: 21 Width: 2 Height: 2 Internal Format: 0x8058
03-27 16:10:05.203 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_9_3dconttex id 21 face 0 level 0: width=2 height=2 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.204 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_9_3dconttex id 21 face 0 level 1: width=1 height=1 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.206 29463 29481 I OGRE    : GLES2Texture::create - Name: VUI3D_10_3dconttex ID: 22 Width: 2 Height: 2 Internal Format: 0x8058
03-27 16:10:05.206 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_10_3dconttex id 22 face 0 level 0: width=2 height=2 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.206 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture VUI3D_10_3dconttex id 22 face 0 level 1: width=1 height=1 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.271 29463 29481 I OGRE    : GLES2Texture::create - Name: blackpix.jpg ID: 23 Width: 1 Height: 1 Internal Format: 0x8051
03-27 16:10:05.272 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture blackpix.jpg id 23 face 0 level 0: width=1 height=1 depth=1 format=PF_B8G8R8
03-27 16:10:05.272 29463 29481 I OGRE    : Texture 'blackpix.jpg': Loading 1 faces(PF_B8G8R8,1x1x1) Internal format is PF_B8G8R8,1x1x1.
03-27 16:10:05.272 29463 29481 I OGRE    : GLES2TextureBuffer::upload - ID: 23 Target: 3553 Format: PF_B8G8R8 Origin format: 0x1907 Data type: 0x1401
03-27 16:10:05.273 29463 29481 I OGRE    : GLES2Texture::create - Name: SO3/Internal/Widgets/BitmapWidget/Texture/Loading ID: 24 Width: 800 Height: 600 Internal Format: 0x8058
03-27 16:10:05.274 29463 29481 I OGRE    : GLES2HardwarePixelBuffer constructed for texture SO3/Internal/Widgets/BitmapWidget/Texture/Loading id 24 face 0 level 0: width=800 height=600 depth=1 format=PF_A8B8G8R8
03-27 16:10:05.277 29463 29481 I OGRE    : GLES2TextureBuffer::upload - ID: 24 Target: 3553 Format: PF_A8B8G8R8 Origin format: 0x1908 Data type: 0x1401
03-27 16:10:05.294 29463 29481 I OGRE    : GLES2TextureBuffer::upload - ID: 24 Target: 3553 Format: PF_A8B8G8R8 Origin format: 0x1908 Data type: 0x1401
03-27 16:10:05.300 29463 29481 I OGRE    : Material scheme switch from Default to ShaderGeneratorDefaultScheme on material : SO3/Internal/Widgets/BitmapWidget/Overlays/Panels/Materials/Loading
03-27 16:10:05.300 29463 29481 I OGRE    : Material scheme switch from ShaderGeneratorDefaultScheme to ShaderGeneratorDefaultScheme on material : SO3/Internal/Widgets/BitmapWidget/Overlays/Panels/Materials/Loading
03-27 16:10:05.311 29463 29481 I OGRE    : Creating resource group 0.mainScene
03-27 16:10:05.312 29463 29481 I OGRE    : Added resource location '/storage/emulated/0/Android/data/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainScene'
03-27 16:10:05.312 29463 29481 I OGRE    : Added resource location '/data/user/0/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainScene'
03-27 16:10:05.313 29463 29481 I OGRE    : [WARNING] ::find Could not find the files in APK with pattern : *
03-27 16:10:05.313 29463 29481 I OGRE    : Added resource location 'APK/partition/' of type 'ScolFileSystem' to resource group '0.mainScene'
03-27 16:10:05.313 29463 29481 I OGRE    : Added resource location 'OBB/partition/' of type 'ScolFileSystem' to resource group '0.mainScene'
03-27 16:10:05.314 29463 29481 I OGRE    : SO3 EXCEPTION(SExceptionInvalidParameters): SSAO technique, not supported by the graphic card! in SSsaoHandler::SeTechniqueParameterValue at D:/dev/scol-technologies/trunk/scol/plugins/so3engine/src/SO3Renderer/SO3SsaoHandler.cpp (line 2664)
03-27 16:10:05.314 29463 29481 I OGRE    : SO3 EXCEPTION(SExceptionInvalidParameters): SSAO technique, not supported by the graphic card! in SSsaoHandler::SeTechniqueParameterValue at D:/dev/scol-technologies/trunk/scol/plugins/so3engine/src/SO3Renderer/SO3SsaoHandler.cpp (line 2664)
03-27 16:10:05.314 29463 29481 I OGRE    : SO3 EXCEPTION(SExceptionInvalidParameters): SSAO technique, not supported by the graphic card! in SSsaoHandler::SeTechniqueParameterValue at D:/dev/scol-technologies/trunk/scol/plugins/so3engine/src/SO3Renderer/SO3SsaoHandler.cpp (line 2664)
03-27 16:10:05.314 29463 29481 I OGRE    : SO3 EXCEPTION(SExceptionInvalidParameters): SSAO technique, not supported by the graphic card! in SSsaoHandler::SeTechniqueParameterValue at D:/dev/scol-technologies/trunk/scol/plugins/so3engine/src/SO3Renderer/SO3SsaoHandler.cpp (line 2664)
03-27 16:10:05.314 29463 29481 I OGRE    : SO3 EXCEPTION(SExceptionInvalidParameters): SSAO technique, not supported by the graphic card! in SSsaoHandler::SeTechniqueParameterValue at D:/dev/scol-technologies/trunk/scol/plugins/so3engine/src/SO3Renderer/SO3SsaoHandler.cpp (line 2664)
03-27 16:10:05.315 29463 29481 I OGRE    : SO3 EXCEPTION(SExceptionInvalidParameters): SSAO technique, not supported by the graphic card! in SSsaoHandler::SeTechniqueParameterValue at D:/dev/scol-technologies/trunk/scol/plugins/so3engine/src/SO3Renderer/SO3SsaoHandler.cpp (line 2664)
03-27 16:10:05.412 29463 29481 I OGRE    : Creating resource group 0.mainScene.scene loader inst
03-27 16:10:05.414 29463 29481 I OGRE    : Added resource location '/storage/emulated/0/Android/data/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainScene.scene loader inst'
03-27 16:10:05.414 29463 29481 I OGRE    : Added resource location '/data/user/0/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainScene.scene loader inst'
03-27 16:10:05.414 29463 29481 I OGRE    : [WARNING] ::find Could not find the files in APK with pattern : *
03-27 16:10:05.414 29463 29481 I OGRE    : Added resource location 'APK/partition/' of type 'ScolFileSystem' to resource group '0.mainScene.scene loader inst'
03-27 16:10:05.414 29463 29481 I OGRE    : Added resource location 'OBB/partition/' of type 'ScolFileSystem' to resource group '0.mainScene.scene loader inst'
03-27 16:10:05.414 29463 29481 I OGRE    : Destroying resource group 0.mainScene.scene loader inst
03-27 16:10:05.415 29463 29481 I OGRE    : Unloading resource group 0.mainScene.scene loader inst
03-27 16:10:05.415 29463 29481 I OGRE    : Finished unloading resource group 0.mainScene.scene loader inst
03-27 16:10:05.415 29463 29481 I OGRE    : Creating resource group 0.mainScene.scene loader inst
03-27 16:10:05.417 29463 29481 I OGRE    : Added resource location '/storage/emulated/0/Android/data/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainScene.scene loader inst'
03-27 16:10:05.417 29463 29481 I OGRE    : Added resource location '/data/user/0/org.imaginer.scol/files/partition/' of type 'ScolFileSystem' to resource group '0.mainScene.scene loader inst'
03-27 16:10:05.417 29463 29481 I OGRE    : [WARNING] ::find Could not find the files in APK with pattern : *
03-27 16:10:05.417 29463 29481 I OGRE    : Added resource location 'APK/partition/' of type 'ScolFileSystem' to resource group '0.mainScene.scene loader inst'
03-27 16:10:05.417 29463 29481 I OGRE    : Added resource location 'OBB/partition/' of type 'ScolFileSystem' to resource group '0.mainScene.scene loader inst'
03-27 16:10:07.551 29463 29481 I OGRE    : Warning: No emissive in wuson0 Material.
03-27 16:10:07.551 29463 29481 I OGRE    : Warning: No transparent in wuson0 Material.
03-27 16:10:07.552 29463 29481 I OGRE    : Added resource location '/data/user/0/org.imaginer.scol/cache/Cache/tmp/download' of type 'FileSystem' to resource group '0.mainScene.scene loader inst'
03-27 16:10:07.601 29463 29481 I OGRE    : Material scheme converted from Default to ShaderGeneratorDefaultScheme on material : wuson0 with group : 0.mainScene.scene loader inst
03-27 16:10:07.618 29463 29481 E OGRE    : Error: Cached function not found SGX_Light_Point_Diffuse
03-27 16:10:07.618 29463 29481 E OGRE    : Error: Cached function not found SGX_Light_Directional_Diffuse
03-27 16:10:07.618 29463 29481 E OGRE    : Error: Cached function not found SGX_Light_Directional_Diffuse
03-27 16:10:07.618 29463 29481 E OGRE    : Error: Cached function not found SGX_Light_Spot_Diffuse
03-27 16:10:07.619 29463 29481 E OGRE    : Error: Cached function not found SGX_Light_Spot_Diffuse
03-27 16:10:07.620 29463 29481 E OGRE    : Error: Cached function not found FFP_Add
03-27 16:10:07.622 29463 29481 I OGRE    : Material scheme switch from ShaderGeneratorDefaultScheme to ShaderGeneratorDefaultScheme on material : wuson0
03-27 16:10:07.682 29463 29481 E OGRE    : Vertex Program:e09abeb5bfb6bdb44865185618a42625_VS Fragment Program:f2ab8117d9246efbe8f3cbdde2ec15c2_FS
03-27 16:10:07.682 29463 29481 E OGRE    : GLSL vertex program result : 
03-27 16:10:07.682 29463 29481 E OGRE    : Vertex info
03-27 16:10:07.682 29463 29481 E OGRE    : -----------
03-27 16:10:07.682 29463 29481 E OGRE    : 0(91) : error C3002: call to undefined function "void FFP_Transform(mat4, vec4, vec4);"
03-27 16:10:07.682 29463 29481 E OGRE    : 0(98) : error C3002: call to undefined function "void SGX_Light_Point_Diffuse(vec3, vec3, vec3, vec4, vec3, vec3, vec3);"
03-27 16:10:07.682 29463 29481 E OGRE    : 0(99) : error C3002: call to undefined function "void SGX_Light_Directional_Diffuse(vec3, vec3, vec3, vec3, vec3);"
03-27 16:10:07.682 29463 29481 E OGRE    : 0(100) : error C3002: call to undefined function "void SGX_Light_Directional_Diffuse(vec3, vec3, vec3, vec3, vec3);"
03-27 16:10:07.682 29463 29481 E OGRE    : 0(101) : error C3002: call to undefined function "void SGX_Light_Spot_Diffuse(vec3, vec3, vec3, vec3, vec4, vec3, vec3, vec3, vec3);"
03-27 16:10:07.682 29463 29481 E OGRE    : 0(102) : error C3002: call to undefined function "void SGX_Light_Spot_Diffuse(vec3, vec3, vec3, vec3, vec4, vec3, vec3, vec3, vec3);"
03-27 16:10:07.684 29463 29481 E OGRE    : Vertex Program:e09abeb5bfb6bdb44865185618a42625_VS Fragment Program:f2ab8117d9246efbe8f3cbdde2ec15c2_FS
03-27 16:10:07.684 29463 29481 E OGRE    : GLSL fragment program result : 
03-27 16:10:07.684 29463 29481 E OGRE    : Fragment info
03-27 16:10:07.684 29463 29481 E OGRE    : -------------
03-27 16:10:07.684 29463 29481 E OGRE    : 0(36) : error C3002: call to undefined function "void FFP_Add(vec3, vec3, vec3);"

--- a/Components/RTShaderSystem/include/OgreShaderFunctionAtom.h
+++ b/Components/RTShaderSystem/include/OgreShaderFunctionAtom.h
@@ -30,6 +30,7 @@ THE SOFTWARE.
 #include "OgreShaderPrerequisites.h"
 #include "OgreStringVector.h"
 #include "OgreGpuProgramParams.h"
+#include "OgreShaderParameter.h"
 
 namespace Ogre {
 namespace RTShader {

--- a/Components/RTShaderSystem/src/OgreShaderFunctionAtom.cpp
+++ b/Components/RTShaderSystem/src/OgreShaderFunctionAtom.cpp
@@ -296,6 +296,15 @@ bool FunctionInvocation::operator < ( const FunctionInvocation& rhs ) const
     return FunctionInvocationLessThan()(*this, rhs);
 }
 
+static uchar getSwizzledSize(const Operand& op)
+{
+    auto gct = op.getParameter()->getType();
+    if (op.getMask() == Operand::OPM_ALL)
+        return GpuConstantDefinition::getElementSize(gct, false);
+
+    return Operand::getFloatCount(op.getMask());
+}
+
 bool FunctionInvocation::FunctionInvocationLessThan::operator ()(FunctionInvocation const& lhs, FunctionInvocation const& rhs) const
 {
     // Check the function names first
@@ -341,25 +350,8 @@ bool FunctionInvocation::FunctionInvocationLessThan::operator ()(FunctionInvocat
         if (itLHSOps->getSemantic() > itRHSOps->getSemantic())
             return false;
 
-        GpuConstantType leftType    = itLHSOps->getParameter()->getType();
-        GpuConstantType rightType   = itRHSOps->getParameter()->getType();
-        
-        // If a swizzle mask is being applied to the parameter, generate the GpuConstantType to
-        // perform the parameter type comparison the way that the compiler will see it.
-        if ((itLHSOps->getFloatCount(itLHSOps->getMask()) > 0) ||
-           (itRHSOps->getFloatCount(itRHSOps->getMask()) > 0))
-        {
-            if (itLHSOps->getFloatCount(itLHSOps->getMask()) > 0)
-            {
-                leftType = (GpuConstantType)((itLHSOps->getParameter()->getType() - itLHSOps->getParameter()->getType()) +
-                                             itLHSOps->getFloatCount(itLHSOps->getMask()));
-            }
-            if (itRHSOps->getFloatCount(itRHSOps->getMask()) > 0)
-            {
-                rightType = (GpuConstantType)((itRHSOps->getParameter()->getType() - itRHSOps->getParameter()->getType()) +
-                                             itRHSOps->getFloatCount(itRHSOps->getMask()));
-            }
-        }
+        uchar leftType    = getSwizzledSize(*itLHSOps);
+        uchar rightType   = getSwizzledSize(*itRHSOps);
 
         if (leftType < rightType)
             return true;
@@ -393,36 +385,7 @@ bool FunctionInvocation::FunctionInvocationCompare::operator ()(FunctionInvocati
         if (itLHSOps->getSemantic() != itRHSOps->getSemantic())
             return false;
 
-        GpuConstantType leftType    = itLHSOps->getParameter()->getType();
-        GpuConstantType rightType   = itRHSOps->getParameter()->getType();
-        
-        if (Ogre::Root::getSingletonPtr()->getRenderSystem()->getName().find("OpenGL ES 2") != String::npos)
-        {
-            if (leftType == GCT_SAMPLER1D)
-                leftType = GCT_SAMPLER2D;
-
-            if (rightType == GCT_SAMPLER1D)
-                rightType = GCT_SAMPLER2D;
-        }
-
-        // If a swizzle mask is being applied to the parameter, generate the GpuConstantType to
-        // perform the parameter type comparison the way that the compiler will see it.
-        if ((itLHSOps->getFloatCount(itLHSOps->getMask()) > 0) ||
-           (itRHSOps->getFloatCount(itRHSOps->getMask()) > 0))
-        {
-            if (itLHSOps->getFloatCount(itLHSOps->getMask()) > 0)
-            {
-                leftType = (GpuConstantType)((itLHSOps->getParameter()->getType() - itLHSOps->getParameter()->getType()) +
-                                             itLHSOps->getFloatCount(itLHSOps->getMask()));
-            }
-            if (itRHSOps->getFloatCount(itRHSOps->getMask()) > 0)
-            {
-                rightType = (GpuConstantType)((itRHSOps->getParameter()->getType() - itRHSOps->getParameter()->getType()) +
-                                             itRHSOps->getFloatCount(itRHSOps->getMask()));
-            }
-        }
-
-        if (leftType != rightType)
+        if (getSwizzledSize(*itLHSOps) != getSwizzledSize(*itRHSOps))
             return false;
     }
 

--- a/Tests/Components/RTShaderSystemTests.cpp
+++ b/Tests/Components/RTShaderSystemTests.cpp
@@ -33,6 +33,8 @@ THE SOFTWARE.
 #include "OgreShaderFFPTransform.h"
 #include "OgreShaderFFPColour.h"
 
+#include "OgreShaderFunctionAtom.h"
+
 using namespace Ogre;
 
 struct RTShaderSystem : public RootWithoutRenderSystemFixture
@@ -109,4 +111,23 @@ TEST_F(RTShaderSystem, TargetRenderState)
 
     EXPECT_TRUE(pass->hasGpuProgram(GPT_VERTEX_PROGRAM));
     EXPECT_TRUE(pass->hasGpuProgram(GPT_FRAGMENT_PROGRAM));
+}
+
+TEST_F(RTShaderSystem, FunctionInvocationOrder)
+{
+    using namespace RTShader;
+
+    FunctionInvocation a("name", 0);
+    FunctionInvocation b("name", 0);
+    FunctionInvocation c("name", 0);
+
+    a.pushOperand(ParameterFactory::createConstParam(Vector3()), Operand::OPS_IN);
+    b.pushOperand(ParameterFactory::createConstParam(Vector3()), Operand::OPS_IN, Operand::OPM_XY);
+    c.pushOperand(ParameterFactory::createConstParam(Vector3()), Operand::OPS_IN, Operand::OPM_XYZ);
+
+    EXPECT_FALSE(b == a);
+    EXPECT_TRUE(b < a);
+
+    EXPECT_TRUE(c == a);
+    EXPECT_FALSE(c < a);
 }

